### DISCUSSION TITLE
How to step over with debugger

### DISCUSSION THREAD

**Discussion by HenningOI** [pamidur/aspect-injector#195] [Q&A]
Dear developers, first thanks for this nice AOP component :)

I know this question is more about visual studio debugger, but maybe you have some ideas.

I tried [DebugStepThrough] to avoid stepping into my aspect while debugging, but when I call an interface function of the Instance, the debugger steps into that function anyway.
I don't want to write this StepThrough Attribute on every class I implement that method, so do you know any way how to avoid that?
Thanks a lot in advance
Lumo

**Comment by pamidur** [comment]
Hi @Lumoryel 
If Just-My-Code is disabled the debugger ill stop in method marked with this attribute. Per https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debuggerstepthroughattribute?view=net-6.0

If you want to completely hide it from debugget you may want to use https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debuggerhiddenattribute?view=net-6.0
